packages/react-components/react-utilities/src/compose/isSlot.ts

```diff
@@ -0,0 +1,62 @@
+import { isValidElement } from 'react';
+import type { ExtractSlotProps, Slot, SlotShorthandValue, UnknownSlotProps } from './types';
+
+/**
+ * Guard method that validates if a shorthand is a slot
+ * can be used to extends properties provided by a slot
+ *
+ * @example
+ * ```
+ * const backdropSlot = resolveShorthand(backdrop, {
+ *  defaultProps: {
+ *    onClick: useEventCallback(event => {
+ *     if (isSlot(backdrop)) {
+ *        backdrop.onClick?.(event)
+ *      }
+ *      // do something after passing click down the line
+ *    }),
+ *  },
+ * })
+ * ```
+ * @example
+ * ```
+ * const handleBackDropClick = (event) => {
+ *  // do your thing
+ * }
+ * const backdropSlot = resolveShorthand(backdrop, {
+ *  defaultProps: {
+ *    onClick: useEventCallback(mergeCallbacks(isSlot(backdrop) ? backdrop.onClick : undefined, handleBackdropClick))
+ * })
+ * ```
+ */
+export function isSlot<Shorthand extends Slot<UnknownSlotProps>>(
```

Comment:
IMO `isSlot` is not perfect name for this, I suggest to keep terminology and use `isResolvedShorthandValue()`. Which is more verbose, but means that it means 🐱

Reply:
I don't get it, isn't the term slot good enough to describe a resolved shorthand?